addOns/encoder/src/test/java/org/zaproxy/addon/encoder/processors/predefined/JavaScriptStringDecoderUnitTest.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2022 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.encoder.processors.predefined;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+import org.junit.jupiter.api.Test;
+import org.zaproxy.addon.encoder.processors.EncodeDecodeResult;
+
+public class JavaScriptStringDecoderUnitTest extends ProcessorTests<JavaScriptStringDecoder> {
+
+    @Override
+    protected JavaScriptStringDecoder createProcessor() {
+        return JavaScriptStringDecoder.getSingleton();
+    }
+
+    @Test
+    void shouldEncodeSimpleString() throws Exception {
+        // Given / When
+        EncodeDecodeResult result = processor.process("admin");
+        // Then
+        assertThat(result.hasError(), is(equalTo(false)));
+        assertThat(result.getResult(), is(equalTo("admin")));
+    }
+
+    @Test
+    void shouldEncodeScriptTag() throws Exception {
```

Comment:
This does not seem to add much. (I think I saw in other places some tests that don't seem to be needed, but just skimming.)

Reply:
I don't think we need to be consistent, the processors are not all of the same "type".

Reply:
True